analytics/src/main/java/com/segment/analytics/SegmentIntegration.java

```diff
@@ -334,15 +334,17 @@ void submitFlush() {
       return;
     }
 
-    networkExecutor.submit(
-        new Runnable() {
-          @Override
-          public void run() {
-            synchronized (flushLock) {
-              performFlush();
+    if (!networkExecutor.isShutdown()) {
```

Comment:
to minimize nesting, we can copy the pattern from above:

```
if (networkExecutor.isShutdown()) {
  return;
}

networkExecutor.submit ...
```

Maybe we should emit a log when this happens?

Reply:
ok, will do!